session/object.go

```diff
@@ -0,0 +1,166 @@
+package session
+
+import (
+	"github.com/nspcc-dev/neofs-api-go/v2/session"
+	"github.com/nspcc-dev/neofs-sdk-go/object/address"
+)
+
+// ObjectContext represents NeoFS API v2-compatible
+// context of the object session.
+//
+// It is a wrapper over session.ObjectSessionContext
+// which allows abstracting from details of the message
+// structure.
+type ObjectContext session.ObjectSessionContext
+
+// NewObjectContext creates and returns blank ObjectContext.
+//
+// Defaults:
+//  - not bound to any operation;
+//  - nil object address.
+func NewObjectContext() *ObjectContext {
+	v2 := new(session.ObjectSessionContext)
+
+	return ObjectContextFromV2(v2)
+}
+
+// ObjectContextFromV2 wraps session.ObjectSessionContext
+// into ObjectContext.
+func ObjectContextFromV2(v *session.ObjectSessionContext) *ObjectContext {
+	return (*ObjectContext)(v)
+}
+
+// ToV2 converts ObjectContext to session.ObjectSessionContext
+// message structure.
+func (x *ObjectContext) ToV2() *session.ObjectSessionContext {
+	return (*session.ObjectSessionContext)(x)
+}
+
+// ApplyTo specifies which object the ObjectContext applies to.
+func (x *ObjectContext) ApplyTo(id *address.Address) {
+	v2 := (*session.ObjectSessionContext)(x)
+
+	v2.SetAddress(id.ToV2())
+}
+
+// Address returns identifier of the object
+// to which the ObjectContext applies.
+func (x *ObjectContext) Address() *address.Address {
+	v2 := (*session.ObjectSessionContext)(x)
+
+	return address.NewAddressFromV2(v2.GetAddress())
+}
+
+func (x *ObjectContext) forVerb(v session.ObjectSessionVerb) {
+	(*session.ObjectSessionContext)(x).
+		SetVerb(v)
+}
+
+func (x *ObjectContext) isForVerb(v session.ObjectSessionVerb) bool {
+	return (*session.ObjectSessionContext)(x).
+		GetVerb() == v
+}
```

Comment:
What about having an interface like the one in the API?

Reply:
the approach was taken from `ContainerSessionContext`